Document the resource quota working and steps require to enforce  resource quota on ephemeral storage (#35011)
kubernetes/website: content/en/docs/concepts/configuration/manage-resources-containers.md

from local ephemeral storage.
{{< /caution >}}

{{< note >}}
To make the resource quota work on ephemeral-storage, two things need to be done:

* An admin sets the resource quota for ephemeral-storage in a namespace.
* A user needs to specify limits for the ephemeral-storage resource in the Pod spec.

If the user doesn't specify the ephemeral-storage resource limit in the Pod spec,

> the resource quota is not enforced on ephemeral-storage.

Reviewer: Are LimitRanges relevant here? (Can you use a LimitRange to enforce a limit on ephemeral storage, even when people who define Deployments, Jobs etc do not specify any limit).

Author: Hi @sftim For more clarification on this  could you please take a look at the comments here.
https://github.com/kubernetes/kubernetes/issues/110956#issuecomment-1216244886

Thanks!!

from local ephemeral storage.
{{< /caution >}}

{{< note >}}
This is an exception for enforcing resource quota on ephemeral-storage.

> 

Reviewer 2: these two lines can be safely dropped?
It is confusing what is an exception.

Reviewer 3: You can see the preview at here:
https://deploy-preview-35011--kubernetes-io-main-staging.netlify.app/docs/concepts/configuration/manage-resources-containers/

> To make the resource quota work on ephemeral-storage, two things need to be done.

Reviewer 4: ```suggestion
To make the resource quota work on ephemeral-storage, two things need to be done:
```

Author: This is Done.

> * an admin sets the resource quota for ephemeral-storage

Reviewer 4: ```suggestion
* An admin sets the resource quota for ephemeral-storage
```

> * An admin sets the resource quota for ephemeral-storage

Reviewer 2: ```suggestion
* An admin sets the resource quota for ephemeral-storage in a namespace.
```

> * User needs to mention the ephemeral-storage resource limit in the Pod spec.

Reviewer 4: ```suggestion
* A user needs to mention the ephemeral-storage resource limit in the Pod spec.
```

> * A user needs to mention the ephemeral-storage resource limit in the Pod spec.

Reviewer 2: ```suggestion
* A user needs to specify limits for the ephemeral-storage resource in the Pod spec.
```

> If the user doesn't specify the ephemeral-storage resource limit in the pod spec,

Reviewer 2: ```suggestion
If the user doesn't specify the ephemeral-storage resource limit in the Pod spec,
```

the resource quota is not enforced on ephemeral-storage. Thus the user can use
any amount of ephemeral-storage.

> 

Reviewer: Am I right that:
this detail about ResourceQuota would apply to all resources, and not just to ephemeral storage?

If so, let's reframe it as a general detail rather than a specific one.

Author: @sftim when we enforce resourceQuota on ephemeral-storage it shows different behaviour and  to be precise it doesn't enforce resouce limits/request  and that is not mention  in documentation. This PR document what steps should we follow if we want to enforce resource quota on ephemeral-storage.

> the resource quota is not enforced on ephemeral-storage. Thus the user can use
> any amount of ephemeral-storage.
> 

Reviewer 2: ```suggestion
the resource quota is not enforced on ephemeral-storage.
```
